www/src/components/CustomThemesForm.tsx

```diff
@@ -0,0 +1,190 @@
+    name: string;
+    urls: string[];
+  } | null;
+  onSave: (theme: { name: string; urls: string[] }) => void;
+}
+
+export interface CustomThemesFormRef {
+  submitForm: () => void;
+}
+
+interface FormData {
+  name: string;
+  urls: { url: string }[];
+}
+
+function isValidCssUrl(url: string) {
+  try {
+    const u = new URL(url);
+    return u.protocol.startsWith('http') && u.pathname.endsWith('.css');
+  } catch {
+    return false;
+  }
+}
+
+const CustomThemesForm = forwardRef<CustomThemesFormRef, CustomThemesFormProps>(({ initialTheme, onSave }, ref) => {
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+    trigger,
+    getValues,
+  } = useForm<FormData>({
+    defaultValues: {
+      name: initialTheme?.name || '',
+      urls: initialTheme?.urls.length ? initialTheme.urls.map(url => ({ url })) : [{ url: '' }],
+    },
+    mode: 'onTouched',
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: 'urls',
+  });
+
+  const urlInputRefs = useRef<(HTMLInputElement | null)[]>([]);
+  const prevUrlsLength = useRef(fields.length);
+
+  useEffect(() => {
+    if (fields.length > prevUrlsLength.current) {
+      urlInputRefs.current[fields.length - 1]?.focus();
+    }
+    prevUrlsLength.current = fields.length;
+  }, [fields.length]);
+
+  const onSubmit = (data: FormData) => {
+    onSave({
+      name: data.name.trim(),
+      urls: data.urls.map(u => u.url.trim()).filter(url => url.length > 0),
+    });
+  };
+
+  const submitForm = async () => {
+    const isValid = await trigger();
+    if (isValid) {
+      const data = getValues();
+      onSubmit(data);
+    }
+  };
+
+  useImperativeHandle(ref, () => ({
+    submitForm,
+  }));
+
+  const handleAddUrl = () => {
+    append({ url: '' });
+  };
+
+  const handleRemoveUrl = (index: number) => {
+    remove(index);
+  };
+
+  return (
+    <Form id="customThemesForm" onSubmit={handleSubmit(onSubmit)}>
+      <Form.Text className="mb-3">
+        Add a custom theme name and one or more CSS URLs to apply your own theme. The CSS files should be
+        accessible via URLs and must end with <code>.css</code>.
+      </Form.Text>
+      <Form.Group controlId="customThemeName" isInvalid={!!errors.name}>
+        <Form.Label>Theme Name</Form.Label>
+        <Form.Control
+          size="sm"
+          {...register('name', {
+            required: 'Theme name is required.',
+            validate: (value) => value.trim().length > 0 || 'Theme name is required.',
+          })}
+        />
+        {errors.name && (
+          <Form.Control.Feedback type="invalid">
+            {errors.name.message}
+          </Form.Control.Feedback>
+        )}
+      </Form.Group>
+      {fields.map((field, idx) => (
+        <Form.Group
+          key={field.id}
+          controlId={`customThemeUrl${idx}`}
+          isInvalid={!!errors.urls?.[idx]}
+        >
+          {idx === 0 && (
+            <Form.Label>CSS URL</Form.Label>
```

Comment:
Maybe

```suggestion
            <Form.Label>CSS URL(s)</Form.Label>
```

Reply:
That sounds like a good plan. It would likely remove the need for some of the logic checking for `idx` being `0` throughout the map too.

Reply:
Will be updated in my next commit, but here's where I landed on this change visually:

<img width="783" alt="image" src="https://github.com/user-attachments/assets/2628dcb6-1efb-44e1-9d57-6f0807feb633" />

Renders a display-only `<Form.Label>` for "CSS URL(s)" and then each of the url inputs define `aria-label="CSS URL"` as a label for the singular url input.

Clicking the `<Form.Label>` will still put focus into the first url input.

Reply:
[update] I changed this once more locally; ultimately settled on `outline-primary` :)